programs/quarter_board/main.c

```diff
@@ -0,0 +1,182 @@
+#include "ch.h"
+#include "hal.h"
+
+#include <icu_lld.h>
+#include "chprintf.h"
+#include <inttypes.h>
+
+#include "communication.h"
+
+#define MAX_PULSE_LENGTH 3000
+#define MIN_PULSE_LENGTH 400
+#define NOT_SEEING_VALUE 3000
+#define QUANTITY_OF_SENSORS 2
+
+#define UNUSED(arg) (void)(arg)
+
+int32_t sensors_values[QUANTITY_OF_SENSORS];
+
+thread_t *commanderp;
+thread_t *ir_measurementp;
+
+static void read_pulse(ICUDriver *icup) {
+    uint8_t i;
+    int32_t value;
+
+    if (icup == &ICUD1) {
+        i = 0;
+    } else {
+        i = 1;
+    }
+    
+    value = icuGetWidthX(icup);
+
+    if (value < MAX_PULSE_LENGTH && value > MIN_PULSE_LENGTH) {    

```

Comment:
Please, reverse the comparison order. This can confuse you (because you expect the value to be first checked with the minimum and then maximum) so it looks like it should never be true... Confusing to have `MAX` before `MIN`.